premailer/premailer.py

```diff
@@ -535,7 +557,12 @@ def _style_to_basic_html_attributes(self, element, style_content,
             elif key == 'vertical-align':
                 attributes['valign'] = value.strip()
             elif key == 'background-color':
-                attributes['bgcolor'] = value.strip()
+                try:
+                    sixColour = self.six_colour(value.strip())
+                except TransparentIsNotAColour:
+                    pass
```

Comment:
oh. I think I see. It's because you have an `else:` next. 
Here's what I suggest you do instead. 

1) Remove the `transparent` check from the `six_colour` function. That way, that function does 1 thing and 1 thing only. 
2) Remove the `TransparentIsNotAColour` exception class
3) Simply do this instead of using exceptions:
```python
if 'transparent' not in value.lower():
    # Only add the 'bgcolor' attribute if the value doesn't contain the word "transparent"
    # and before we add it possibly correct 3 character shortenings to 6 character
    # equivalents. E.g. convert "abc" to "aabbcc".
    attributes['bgcolor'] = self.six_colour(value.strip())
```